### PR TITLE
Add axios and refactor commerce to use that for HTTP

### PR DESCRIPTION
**Pull request by ScopeyNZ**
Also updates cart tests to assert axios is called with expected arguments

All the other feature tests need updating, and I want to update the commerce test suite too... I'll do PRs for those tomorrow though

### CODE REVIEW

File: src/features/checkout.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ class Checkout {
       'GET',
       data,
       callback,
-      error
+      error,
```
Comment (robbieaverill) [inline src/features/checkout.js]: These methods don't have `this.id().then(` prefixes like the cart changes do - is there a reason for that?

File: src/features/cart.js [hunk 2]
```diff
@@ -6,115 +6,80 @@ class Cart {
...
-  update(lineId, data, callback, error) {
-    return this.commerce.request(
-      `carts/${this.id()}/items/${lineId}`,
-      'PUT',
-      data,
-      callback,
-      error
-    );
+  delete() {
+    return this.id().then(id => this.commerce.request(`carts/${id}`, 'delete'));
```
[the remark this answers is not in the document]
Comment (ScopeyNZ) [reply src/features/cart.js]: Yeah I can understand the un-ease. Wrapping into into the `Commerce` object wouldn't necessarily make sense as it shouldn't be concerned with the cart but I think abstracting a `cartRequest` method or something makes sense here. I'll do that.

File: src/features/checkout.js (hunk 1 above)
Comment (ScopeyNZ) [reply src/features/checkout.js]: Only the cart uses this internally resolved `cartId` variable - it's a special case.

Comment (robbieaverill) [reply src/features/checkout.js]: Right - sounds good

File: src/features/tests/cart-test.js [hunk 3]
```diff
@@ -1,12 +1,14 @@
 /* global jest, describe, it, expect, beforeEach */
+jest.mock('axios');
+jest.mock('../../commerce');
 
 import Cart from '../cart';
 import Commerce from '../../commerce';
+import axios from 'axios';
 
-jest.mock('../../commerce');
+const realCommerce = jest.requireActual('../../commerce').default;
```
[the remark this answers is not in the document]
Comment (ScopeyNZ) [reply src/features/tests/cart-test.js]: I can do this. It's the class import that needs renaming though. `mockCommerce` already exists for instances of the class mock (`Commerce`). Should I do `import MockCommerce from '../commerce'` and then rename existing `mockCommerce` variables to `mockCommerceInstance` or something?

[the remark this answers is not in the document]
Comment (ScopeyNZ) [reply src/features/tests/cart-test.js]: Nevermind, forgot JS is case sensitive.

File: src/features/cart.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ScopeyNZ) [reply src/features/cart.js]: This cleaned it up quite a bit actually - nice suggestion. I'm starting to think that we should provide a way to instantiate a cart with a specific cart ID in the future though - or allow a developer to force a new cart for some kind of multi-cart flow.

File: src/features/cart.js [hunk 4]
```diff
@@ -6,116 +6,91 @@ class Cart {
...
-      'DELETE',
-      null,
-      callback,
-      error
-    );
+  update(lineId, data) {
+    return this.request(`items/${lineId}`, 'put', data);
+  }
+
+  contents() {
```
Comment (robbieaverill) [inline src/features/cart.js]: JSDocs woulds be nice, but not blocking

**Comment by robbieaverill** [comment]
Also some examples would be good in the readme, but also not blocking

**Pull request merged**